### PR TITLE
🐛 fixes for tests

### PR DESCRIPTION
**Pull request by mevansgh1**
* update include-controller.test - add tests for Darwin
 * update cleanup-controller.test - reuse stubs

### CODE REVIEW

File: docs-markdown/src/test/suite/controllers/cleanup/cleanup-controller.test.ts [hunk 1]
```diff
@@ -23,12 +23,16 @@ const filePath = resolve(__dirname, "../../../../../../src/test/data/repo/articl
 const folderPath = resolve(__dirname, "../../../../../../src/test/data/repo/articles");
 
 suite("Cleanup Controller", () => {
+    suiteSetup(async () => {
```
Comment (mevansgh1) [inline docs-markdown/src/test/suite/controllers/cleanup/cleanup-controller.test.ts:26]: I don't see any tests that are impacted by from stubbing `sendTelemetryData`.  We  also stub and restore this method 20+ times.  This change creates a single stub which is created once before all tests and restored once when all tests are completed.  It also prevents the tests from throwing the following stack trace during runtime

> stack trace: TypeError: Cannot read property 'sendTelemetryEvent' of undefined
> 	at Object.sendTelemetryData (/Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/helper/telemetry.js:41:26)
> 	at Object.<anonymous> (/Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:162:29)
> 	at Generator.next (<anonymous>)
> 	at /Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:8:71
> 	at new Promise (<anonymous>)
> 	at /Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:4:12
> 	at /Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:67:53
> 	at new Promise (<anonymous>)
> 	at /Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:67:20
> 	at Object._withProgress (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:544:963)
> 	at Object.withProgress (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:544:717)
> 	at Object.withProgress (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:867:255)
> 	at Object.<anonymous> (/Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:55:25)
> 	at Generator.next (<anonymous>)
> 	at fulfilled (/Users/mevans/mygit/vscode-docs-authoring/docs-markdown/out/src/controllers/cleanup/cleanup-controller.js:5:58)
> extensionHostProcess.js:920

File: docs-markdown/src/test/suite/controllers/include-controller.test.ts [hunk 2]
```diff
@@ -14,14 +16,13 @@ const expect = chai.expect;
 
 const root = workspace.workspaceFolders![0].uri;
 const testFile = "../../../../../src/test/data/repo/articles/includes.md";
+const testFilePath = resolve(__dirname, testFile);
 const qpSelectionItems = [
-    { description: root.fsPath + "\\includes", label: "1.md" },
-    { description: root.fsPath + "\\includes", label: "2.md" },
-    { description: root.fsPath + "\\includes", label: "3.md" },
+    { description: root.fsPath + sep + "includes", label: "1.md" },
```
Comment (mevansgh1) [inline docs-markdown/src/test/suite/controllers/include-controller.test.ts:21]: Using `path.sep` makes paths OS agnostic.

File: docs-markdown/src/test/suite/controllers/include-controller.test.ts [hunk 3]
```diff
@@ -50,81 +55,88 @@ suite("Include Controller", () => {
...
+        stub.restore();
     });
-    test("Window NT - includeOneFileEmptyLine", async () => {
-        const editor = window.activeTextEditor!;
-        moveCursor(editor, 12, 0);
-        await insertBlankLine(editor);
-        moveCursor(editor, 12, 0); // move cursor back
-        await sleep(extendedSleepTime);
-        window.showQuickPick = () => {
+    test("Window NT - insertInclude", async () => {
```
Comment (mevansgh1) [inline docs-markdown/src/test/suite/controllers/include-controller.test.ts:73]: These 3 "Windows NT" tests are failing *nix systems (tested on Ubuntu & Mac OS), i.e. 

>   1) Include Controller
>        Window NT - includeOneFileEmptyLine:
> 
>       AssertionError: expected '[!INCLUDE [1](../../repo/includes/1.md)]' to equal '[!INCLUDE [1](../includes/1.md)]'
>       + expected - actual
> 
>       -[!INCLUDE [1](../../repo/includes/1.md)]
>       +[!INCLUDE [1](../includes/1.md)]
>       

I refactored these tests and added a test specific to each arche type `Windows_NT` and `Darwin`.

File: docs-markdown/src/test/suite/controllers/include-controller.test.ts (hunk 2 above)
Comment (bharney0) [reply docs-markdown/src/test/suite/controllers/include-controller.test.ts:21]: good to know! I'll have to keep that in mind for future changes.

**Comment by bharney0** [comment]
Thank you for the contribution to Docs Authoring Pack. These changes look good! And the fixes worked for removing the errors with the rejected promise for the telemetryData stub.

**Pull request merged**